### ISSUE TITLE
TBsyn not conecting to server

### ISSUE THREAD

**Issue by TBrianB** [jobisoft/EAS-4-TbSync#259]
## Your environment Windows 10 PC and laptop

TbSync version: 4.8
EAS-4-TbSync version: 4.8
Thunderbird version: 115.6.1

## Actual behavior
..Contqcting server refused

## Steps to reproduce
...On laptop -Tried reenetering password and server several times
Tried deleting account and recreating it
Tried reinstalling TBSync and EASync

Tried on mymain PC and had the same result.
[Thunderbird.pdf](https://github.com/jobisoft/EAS-4-TbSync/files/14030098/Thunderbird.pdf)

![Screenshot_1](https://github.com/jobisoft/EAS-4-TbSync/assets/143229939/0b410a9e-be1c-4e9a-9f4e-08616d66a292)
Log (PDF) and screenshot included with this report
To help resolving your issue, enable debug logging (TbSync Account Manager -> Help) and send me the debug.log via e-mail (use the title of your issue as subject of the email).

**Comment by alessiot89** [comment]
I have a similar issue for an outlook.com account.

**Comment by TBrianB** [comment]
I do seem to be able to access my outlook.com emails. So they aren't down.

**Comment by alessiot89** [comment]
yes, web mail works as well android outlook app sync (it appears in the outlook.com list of "mobile devices")

**Comment by RobMahn** [comment]
I am getting the same errors on both of my outlook.com accounts.  I have the same version reported above, except on Windows 11.  Since this software has not changed on over a month, I suspect that Microsoft has done something with eas.outlook.com and/or the ActiveSync version v14.0.  About a year ago, they were causing problems with the hotmail.com accounts.

**Comment by Aholicknight** [comment]
I am having the same issue, was working perfectly fine before but when I opened thunderbird today, it says my password is wrong.

![thunderbird_20240118_ZcIgP0jnxA](https://github.com/jobisoft/EAS-4-TbSync/assets/7843719/c373aa17-b558-45cd-90cc-a7528762a7f6)

Hope this gets fixed soon

**Comment by jobisoft** [comment]
If someone could find out what happened to the url and what we should use instead, that would be helpful.

**Comment by Gianni1962** [comment]
Mine has started behaving normally after inserting the password at the popup request.

**Comment by RobMahn** [comment]
Mine magically started working too.  I didn't even have to insert the password again, just sync.  I'm thinking Microsoft had a problem with validating the app password and has since made a fix.

**Comment by TBrianB** [comment]
Annoyingly mine is stil not working. On PC. I'll check my laptop, soon.

**Comment by InFerYes** [comment]
Is this possibly related to these changes by Microsoft?

https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online

**Comment by alessiot89** [comment]
> Annoyingly mine is stil not working. On PC. I'll check my laptop, soon.
same, mine still asking password at every try

**Comment by ColeBantam** [comment]
Same Problem here with two outlook.com accounts :(
I'm afraid InFerYes could be right, and Microsoft is shutting down basic-Auth for Outlook.com?
Does TBSync support modern auth?

**Comment by alessiot89** [comment]
I recreated again the sync using this time the Office 365 option that is based on OAuth 2.0 and this time worked.
Looks like AES ActyveSync is simply rejected not by outlook.com :\

**Comment by ColeBantam** [comment]
> I recreated again the sync using this time the Office 365 option that is based on OAuth 2.0 and this time worked. Looks like AES ActyveSync is simply rejected not by outlook.com :\

That solved it for me too. But MS still seems to have troubles as there are still a lot of connection issues

**Comment by ColeBantam** [comment]
> It did not work on the second account. First it failed completely. Upon another retry, it appears to logon but fails during "Waiting for folder list update". Then it gives, "Communication error (HTTP status 503)".

I had the same issues. As I wrote before, MS is having issues currently ... Simply reboot Thunderbird and try again. One time it will work.

**Comment by TBrianB** [comment]
The 365 worked for me too. I also got the strange message about not getting information from the server, but then it worked.. Didn't have to change anything else.
Also I had to enable cookies, slightly annoying.

**Comment by zakius** [comment]
it really seems MS is breaking app passwords all the time recently, I have this issue of sometimes working and sometimes asking for password in 2 apps for like a week now, fortunately OAuth finally started working for me so I may rely on that now

**Comment by mdear** [comment]
> > Annoyingly mine is stil not working. On PC. I'll check my laptop, soon.
> 
> same, mine still asking password at every try

I'm also stuck.  I tried both available versions of ActiveSync v2.5 and v14.0.  Popup endlessly asking me for my password.

Automatic configuration option not working,  Please confirm if the following server address is correct : 

outlook.office365.com/EWS/Exchange.asmx


 Calendar sync was working until recently.  I'm currently running :

Edition	Windows 11 Home
Version	23H2
Installed on	‎2023-‎05-‎25
OS build	22631.3085
Experience	Windows Feature Experience Pack 1000.22684.1000.0

**Comment by TBrianB** [comment]
You may need the separate, app specific password.I've kind of lost track of  which internet services do  that now and which don't. I think there's a link (above).

**Comment by RobMahn** [comment]
I have two outlook.com accounts to sync.  Prior to last week, I was using app passwords, which are no longer working.  My Microsoft accounts will not let me make a new one any longer.

So, I've been trying to use the Microsoft Office 365 OAuth2.0 connection.

I can only get one account to work at a time.  Here are the problems I've encountered.

1. If I delete an account and restart Thunderbird, the account re-appears.  To get the account to stay deleted, I must follow up after the delete action by selecting another account before the restart. (I also have a Gmail account.)  It turns out deleting the account is not necessary for this test, but I just wanted to note this issue.
2. Adding the first Microsoft Office 365 works.  But, when the account is first enabled, I'm confronted with "The server does not provide information about the supported ActiveSync versions. ...".  Clicking "Try again" always succeeds.  Now the first account is working.
3. Addin the second account makes the same complaint about ActiveSync versions, but clicking "Try again" never works.  If I select v14.0, it proceeds to getting a 503 error.  I assume because it is using an old version.
4. Deleting the accounts and swapping the order of addition has the same outcome with the other account working.
5. Disabling both accounts and enabling only the broken account does not make it work.  But, restarting Thunderbird will then make the broken account work.
6. Then, enabling the other account results in a 503 error.
7. In further testing, you don't even need to disable accounts to show that only one account at a time will work.  If you just restart Thunderbird and select an account to sync, the second account will fail to sync.  Restart again and swap accounts has the same results.

Please look at the initial attempt at finding the supported ActiveSync version and the saved information that appears to cross multiple accounts, causing the second account to sync fail.

**Comment by TBrianB** [comment]
There's clearly more to the problem than just switching to that 365 OAuth nonsense. That seems more like a limited workaround, with the weird error message.
It also seems to have let me in for a f***ing advert in the Windows Outlook.com app now- with the new 365 type log in. I hadn't ever seen these nuisances before.
TBH I can't wait until there's a TB for iPhone and I'll jump ship straight away.
(There is one for Android already I believe).

**Comment by RobMahn** [comment]
> There's clearly more to the problem than just switching to that 365 OAuth nonsense. That seems more like a limited workaround, with the weird error message. It also seems to have let me in for a f***ing advert in the Windows Outlook.com app now- with the new 365 type log in. I hadn't ever seen these nuisances before. TBH I can't wait until there's a TB for iPhone and I'll jump ship straight away. (There is one for Android already I believe).

I don't see that Oauth2 is nonsense.  I believe I have identified the apparent randomness in connectivity, and this should be able to be resolved in TbSync.  In the meantime, if you have multiple accounts, you need to restart Thunderbird to sync them one at a time.

Yes, Microsoft added advertising to Outlook.com, and they said it was coming for some time.  This is another reason to use Thunderbird.  Not part of this conversation.

Thunderbird for Android is not yet released ... [https://blog.thunderbird.net/2023/12/when-will-thunderbird-for-android-be-released/](url).  Also, not part of this conversation.

**Comment by RobMahn** [comment]
The situation that I detailed above has already been classified as a bug in issue #249 "Two Microsoft Office 365 accounts configured, only one can sync" on Dec 2, 2023.

So, I believe that MS is stopping the use of app passwords and everyone will need to start using Oauth2.  For single accounts, this is not a problem.  For multiple accounts, you can only sync one at a time after restarting TB until issue #249 is fixed.

**Comment by InFerYes** [comment]
I can't use the option "Microsoft Office 365", which seems to be the only option with OAuth2, as it says "User not found (HTTP Error 404)". It's also not an Office 365 account, but a plain and simple Microsoft account. We share the calendars in our family, a holdover from the Windows Phone times.

**Comment by 1lutz** [comment]
I have the same problem as @InFerYes. When trying to use the "Microsoft Office 365" Option with my standard MS Account i get "HTTP Error 404". There is nothing logged in the event protocol popup.
Using manual configuration with "outlook.office365.com" as hostname and a newly created app password does not work either. It keeps prompting me for a password in a loop.

**Comment by jobisoft** [comment]
**outlook.de & outlook.com accounts not working**

There seem to be a lot of different issues lately. One thing I identified is that the Microsoft lookup for outlook.com or outlook.de addresses seems to return a wrong value. I have an outlook.de account and this lookup for addr@outlook.de

https://autodiscover-s.outlook.com/autodiscover/autodiscover.json?Email=addr%40outlook.de&Protocol=ActiveSync

returns **eas.outlook.com**, but that does not work to actually connect to the server. But manually changing the server to **outlook.office365.com** works for me (and NOT selecting a specific version, but use "best available")

Note: hotmail addresses do need eas.outlook.com. I have no information about msn.com and live.com addresses

Maybe there is a newer autodiscover url I should use?

**Comment by InFerYes** [comment]
I have a **live.be** address that I use. When I change the server address to `outlook.office365.com` it still fails to authenticate when it prompts for my password. The ActiveSync version was already set to "best available".

**Comment by 1lutz** [comment]
> I have a **live.be** address that I use. When I change the server address to `outlook.office365.com` it still fails to authenticate when it prompts for my password. The ActiveSync version was already set to "best available".

I just wanted to share how i finally managed to connect. Maybe it helps you or others.

1. I am using a MS Account with a **gmail.com** address. In Outlook Web I noticed that there also is an **outlook.com** address available, which was apparently automatically created. I set it as my primary alias.
2. Back in Thunderbird I tried to connect using the "Microsoft Office 365" option and the **outlook.com** alias mentioned before.
3. I got a new popup with the Microsoft specific login page with an error message. But within the popup there was an option to login again. Then I entered my normal **gmail.com** address. The login proceeded successfully.
4. There was just one step missing: Sync failed at first because it used **eas.outlook.com** as server. I had to change it manually to **outlook.office365.com**. Then I finally was able to sync my calendar and contacts.
5. After that I went back to Outlook Web and changed my primary sender address to my normal **gmail.com** address.

Sorry, if my english is bad. I am not a native speaker.

**Comment by BAWLMER** [comment]
FWIW, I have 1 hotmail account that stopped working back in January. I  have tried all the above fixes, in different combinations, with no success. I also have an MSN account that I set up and it synced perfectly using the default EAS settings. It makes no sense. 
Just for grins, I set up my hotmail account to use the built-in OS calendar, and it signed in to the EAS server without a problem. 
The calendar did not sync, but I also received no errors.

Linux Mint 21.3
Thunderbird 115.8.1
TBsync 4.8

[debug.log](https://github.com/jobisoft/EAS-4-TbSync/files/14629116/debug.log)
